homeassistant/components/timer.py

```diff
@@ -0,0 +1,402 @@
+            yield from asyncio.wait(tasks, loop=hass.loop)
+
+    descriptions = yield from hass.async_add_job(
+        load_yaml_config_file, os.path.join(
+            os.path.dirname(__file__), 'services.yaml')
+    )
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_START, async_handler_service,
+        descriptions[DOMAIN][SERVICE_START], SERVICE_SCHEMA_DURATION)
+    hass.services.async_register(
+        DOMAIN, SERVICE_PAUSE, async_handler_service,
+        descriptions[DOMAIN][SERVICE_PAUSE], SERVICE_SCHEMA)
+    hass.services.async_register(
+        DOMAIN, SERVICE_CANCEL, async_handler_service,
+        descriptions[DOMAIN][SERVICE_CANCEL], SERVICE_SCHEMA)
+    hass.services.async_register(
+        DOMAIN, SERVICE_FINISH, async_handler_service,
+        descriptions[DOMAIN][SERVICE_FINISH], SERVICE_SCHEMA)
+
+    yield from component.async_add_entities(entities)
+    return True
+
+
+class Timer(Entity):
+    """Representation of a timer."""
+
+    def __init__(self, hass, object_id, name, icon,
+                 weeks, days, hours, minutes, seconds):
+        """Initialize a timer."""
+        self.entity_id = ENTITY_ID_FORMAT.format(object_id)
+        self._name = name
+        self._state = STATUS_IDLE
+        self._duration = timedelta(weeks=weeks, days=days, hours=hours,
+                                   minutes=minutes, seconds=seconds)
+        self._remaining = self._duration
+        self._icon = icon
+        self._hass = hass
+        self._start = None
+        self._end = None
+        self._listener = None
+
+    @property
+    def should_poll(self):
+        """If entity should be polled."""
+        return True
+
+    @property
+    def name(self):
+        """Return name of the timer."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon to be used for this entity."""
+        return self._icon
+
+    @property
+    def state(self):
+        """Return the current value of the timer."""
+        return self._state
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes."""
+        start = dt_util.as_local(self._start) if self._start else None
+        end = dt_util.as_local(self._end) if self._end else None
+        return {
+            ATTR_DURATION: self._duration.__str__(),
+            ATTR_START: start,
+            ATTR_END: end,
+            ATTR_REMAINING: self._remaining.__str__()
+        }
+
+    @Throttle(UPDATE_INTERVAL)
+    def update(self):
+        """Calculate remaining time."""
+        if self._state == STATUS_ACTIVE:
+            self._remaining = self._end - dt_util.utcnow()
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Call when entity is about to be added to Home Assistant."""
+        # If not None, we got an initial value.
+        if self._state is not None:
+            return
+
+        state = yield from async_get_last_state(self.hass, self.entity_id)
```

Comment:
Refer to `async_get_last_state` via `hass.helpers.restore_state.async_get_last_state(self.entity_id)`

Reply:
That's confusing me. I've searched through other components, and found plenty that do it this way. Is this something that should be changed in the future?
Using `hass` directly works, but for one that get's me a lint-error (undefined variable), and with the needed indents (8 spaces) I'm already at 76 cols and I'd have start a new line just for the entity ID. I'm fine with doing that if you confirm that's really how it should be.

Reply:
This is the new way of doing it. There are still a lot not converted yet but we'll get there one day (or probably not). 

To break it up, you can either wrap it all in parentheses or just get restore_state as a local variable.

```python
(hass.helpers.restore_state.
    async_get_last_state(self.entity_id))

restore_state = hass.helpers.restore_state
restore_state.async_get_last_state(…)
```